src/helpers.ts

```diff
@@ -46,7 +46,7 @@ const deprecatedModelMap = new Map<string, string>([
   // Legacy - gpt-4-turbo is no longer in preview, use the latest model alias
   ['gpt-4-1106-preview', 'gpt-4-turbo'],
   // Legacy - All gpt-3.5-turbo models are now 16k
-  ['gpt-3.5-turbo-16k', 'gpt-3.5-turbo'],
```

Comment:
**Bug** - This code should map `<DEPRECATED MODEL>` -> `<NEWER MODEL>`, new code doesn't make sense.

As a general note, we don't want to remove all mentions of the 16k model because we want users to be gracefully "upgraded" to newer models. Otherwise, users with deprecated models that this extension no longer recognizes could put the extension in a "broken" state. This random breakage would be confusing for the user.

Reply:
Yeah, you're right. Another approach may be just to keep `gpt-3.5-turbo-16k` until `2024-09-13`, which is the date that OpenAI will shut it down, then that's not our problem 😆

Reply:
Good idea. :+1:

Reply:
@Christopher-Hayes it's done now, thanks.

Reply:
Oh no I rebased on the wrong branch...

Reply:
Also conflicted with #85

Reply:
@Christopher-Hayes on 13th Sep, do you think we can use `gpt-3.5-turbo-instruct` to replace `gpt-3.5-turbo-16k`, or should we just remove `gpt-3.5-turbo-16k`?

Reply:
Based on recent news, we maybe should just aim to move all users to `gpt-4o-mini`. It basically does everything better than 3.5-turbo, while being cheaper.

Instruct is meant to be used with the "completions endpoint," which adds some complications like not supporting streaming. We do use the completions endpoint when models require it (i.e. some OpenRouter models), but in this case we don't need that model.

Reply:
Created an issue for that - #93